backend/internal/api/handlers.go

```diff
@@ -0,0 +1,299 @@
+// Package api provides HTTP handlers with repository access
+package api
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/ali01/mnemosyne/internal/config"
+	"github.com/ali01/mnemosyne/internal/models"
+	"github.com/ali01/mnemosyne/internal/repository"
+	"github.com/gin-gonic/gin"
+)
+
+// Handler holds dependencies for HTTP handlers
+type Handler struct {
+	repos *repository.Repositories
+	cfg   *config.Config
+}
+
+// getGraph returns the full graph data from the database
+func (h *Handler) getGraph(c *gin.Context) {
+	ctx := context.Background()
+
+	// Get pagination parameters
+	limit, _ := strconv.Atoi(c.DefaultQuery("limit", "1000"))
```

Comment:
Consider adding input validation for pagination parameters. While strconv.Atoi returns 0 for invalid input (which is safe), you should validate that limit is within reasonable bounds (e.g., 1-10000) to prevent potential performance issues or memory exhaustion.